### DISCUSSION TITLE
How to implement a validation if the queue does not exist when mounting the consumer

### DISCUSSION THREAD

**Discussion by alejandro-jara** [Havret/dotnet-activemq-artemis-client#383] [Q&A]
Hi, I'm new to queuing implementation. I have defined a queue in Artemis that apparently if it doesn't receive new messages in N amount of time, Artemis makes the decision to delete it.
Having the service listening, the connection is interrupted and the service stops working.
Do you recommend that I implement a reconnection in case of failure? and making the retry time be exponential (1st try 20ms, 2nd try 40ms, etc) or that in some way indicate that said queue is never eliminated?
The problem I see with the first option is that probably (because of the particular use case) there can be a long time where you don't need to insert messages in the queue.

**Comment by Havret** [comment]
Hello @alejandro-jara, 

What you're describing is rather unusual, and undesired behavior. Generally speaking all your queues should be durable, so you can reliably consume messages without any data loss. How do you define your queue? It would be perfect if you could provide MRE (https://en.wikipedia.org/wiki/Minimal_reproducible_example) of your producing and consuming code so I could take a look. 

Best,
Havret
